fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -342,57 +408,74 @@ fi
 
 log_Msg "mkdir -p ${fMRIFolder}/MotionCorrection"
 mkdir -p "$fMRIFolder"/MotionCorrection
+
 ${RUN} "$PipelineScripts"/MotionCorrection.sh \
        "$fMRIFolder"/MotionCorrection \
        "$fMRIFolder"/"$NameOffMRI"_gdc \
-       "$fMRIFolder"/"$ScoutName"_gdc \
+       "$fMRIReferenceImage" \
        "$fMRIFolder"/"$NameOffMRI"_mc \
        "$fMRIFolder"/"$MovementRegressor" \
        "$fMRIFolder"/"$MotionMatrixFolder" \
        "$MotionMatrixPrefix" \
-       "$MotionCorrectionType"
+       "$MotionCorrectionType" \
+       "$fMRIReferenceReg"
 
 # EPI Distortion Correction and EPI to T1w Registration
-log_Msg "EPI Distortion Correction and EPI to T1w Registration"
-
 DCFolderName=DistortionCorrectionAndEPIToT1wReg_FLIRTBBRAndFreeSurferBBRbased
 DCFolder=${fMRIFolder}/${DCFolderName}
 
-if [ -e ${DCFolder} ] ; then
-    ${RUN} rm -r ${DCFolder}
+if [ $fMRIReference = "NONE" ] ; then
+  log_Msg "EPI Distortion Correction and EPI to T1w Registration"
+
+  if [ -e ${DCFolder} ] ; then
+      ${RUN} rm -r ${DCFolder}
+  fi
+  log_Msg "mkdir -p ${DCFolder}"
+  mkdir -p ${DCFolder}
+
+  ${RUN} ${PipelineScripts}/DistortionCorrectionAndEPIToT1wReg_FLIRTBBRAndFreeSurferBBRbased.sh \
+         --workingdir=${DCFolder} \
+         --scoutin=${fMRIFolder}/${ScoutName}_gdc \
+         --t1=${T1wFolder}/${T1wImage} \
+         --t1restore=${T1wFolder}/${T1wRestoreImage} \
+         --t1brain=${T1wFolder}/${T1wRestoreImageBrain} \
+         --fmapmag=${MagnitudeInputName} \
+         --fmapphase=${PhaseInputName} \
+         --fmapgeneralelectric=${GEB0InputName} \
+         --echodiff=${deltaTE} \
+         --SEPhaseNeg=${SpinEchoPhaseEncodeNegative} \
+         --SEPhasePos=${SpinEchoPhaseEncodePositive} \
+         --echospacing=${EchoSpacing} \
+         --unwarpdir=${UnwarpDir} \
+         --owarp=${T1wFolder}/xfms/${fMRI2strOutputTransform} \
+         --biasfield=${T1wFolder}/${BiasField} \
+         --oregim=${fMRIFolder}/${RegOutput} \
+         --freesurferfolder=${T1wFolder} \
+         --freesurfersubjectid=${Subject} \
+         --gdcoeffs=${GradientDistortionCoeffs} \
+         --qaimage=${fMRIFolder}/${QAImage} \
+         --method=${DistortionCorrection} \
+         --topupconfig=${TopupConfig} \
+         --ojacobian=${fMRIFolder}/${JacobianOut} \
+         --dof=${dof} \
+         --fmriname=${NameOffMRI} \
+         --subjectfolder=${SubjectFolder} \
+         --biascorrection=${BiasCorrection} \
+         --usejacobian=${UseJacobian}
+else
+    log_Msg "linking EPI distortion correction and T1 registration from ${fMRIReference}"
+    if [ -e ${DCFolder} ] ; then
+        log_Warn "     ... removing stale link (or preexisiting files)"
+        rm -r ${DCFolder}
+    fi
+    ln -s ${fMRIReference}/${DCFolderName} ${DCFolder}
+ 
+    ${FSLDIR}/bin/imcp ${fMRIReference}/${RegOutput} ${fMRIFolder}/Reference${RegOutput}
```

Comment:
Is this `Reference${RegOutput}` file used anywhere?  If not, does it serve a purpose?

Reply:
Well, you suggested it would be good to have it with a name that would better inform what it is. And it is copied in the `${fMRIFolder}` from `${DCFolder}` and I guessed there is a reason for it. Why does `DistortionCorrectionAndEPIToT1wReg_FLIRTBBRAndFreeSurferBBRbased.sh` save it, if it is of no interest?

Reply:
Not everything that gets copied necessarily gets used, and it appears that the `--oregim=${fMRIFolder}/${RegOutput}` flag of `DistortionCorrectionAndEPIToT1wReg_FLIRTBBRAndFreeSurferBBRbased.sh` is in that category.  IMO, we want to copy the minimal number of files necessary as part of this "reference BOLD" addition.

Reply:
Ok. I'll remove what is not needed downstream.

Reply:
It seems `${fMRIFolder}/${JacobianOut}` is not needed either.